src/midst_toolkit/models/clavaddpm/dataset_utils.py

```
@@ -95,11 +100,26 @@ def encode_and_merge_features(
         )
     )
 
+    # Load pre-fitted label encoders from pkl if provided, otherwise fit on current data
+    preloaded_encoders: dict[str, LabelEncoder] | None = None
+    if label_encoders_path is not None:
+        _pkl_path = Path(label_encoders_path)
+        if _pkl_path.exists():
+            with open(_pkl_path, "rb") as _f:
+                preloaded_encoders = pickle.load(_f)
```

Ruff (0.15.14)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)